packages/ui/src/components/va-data-table/hooks/useTableScroll.ts

```diff
@@ -0,0 +1,101 @@
+import { shallowRef, onMounted, computed, onBeforeUnmount, watch, nextTick } from 'vue'
+
+import { useHTMLElement } from '../../../composables'
+
+interface useTableScrollProps {
+  stickyHeader: boolean
+  stickyFooter: boolean
+  height: string | number | undefined
+  scrollTopMargin: number | undefined
+  scrollBottomMargin: number | undefined
+}
+
+export default function useTableScroll (
+  props: useTableScrollProps,
+  emit: (event: 'scroll:bottom' | 'scroll:top', ...args: any[]) => void,
+) {
+  let observer: IntersectionObserver
+
+  const scrollContainer = shallowRef<HTMLDivElement>()
+  const scrollContainerHTML = useHTMLElement(scrollContainer)
+
+  const topTrigger = shallowRef<HTMLDivElement>()
+  const topTriggerHTML = useHTMLElement(topTrigger)
+
+  const bottomTrigger = shallowRef<HTMLDivElement>()
+  const bottomTriggerHTML = useHTMLElement(bottomTrigger)
+
+  const isScrollable = computed(() => !!(props.height))
```

Comment:
Can't table be scrollable without `height` prop? I mean, smby can set table height with CSS. At least, I've made it.